services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/helpers/thread/PortBlinker.groovy

```diff
@@ -0,0 +1,49 @@
+package org.openkilda.functionaltests.helpers.thread
+
+import org.openkilda.testing.model.topology.TopologyDefinition.Switch
+import org.openkilda.testing.service.northbound.NorthboundService
+
+class PortBlinker extends Thread {
+    private NorthboundService northbound
+    private volatile boolean running
+
+    final Switch sw
+    final int port
+    Date timeStarted
+    Date timeStopped
+
+    PortBlinker(NorthboundService northbound, Switch sw, int port) {
+        super()
+        this.northbound = northbound
+        this.sw = sw
+        this.port = port
+    }
+
+    @Override
+    synchronized void start() {
+        super.start()
+        timeStarted = new Date()
+    }
+
+    @Override
+    void run() {
+        running = true
+        while(running) {
+            northbound.portDown(sw.dpId, port)
```

Comment:
It may make sense to issue portDown and portUp asynchronously, so that minimum delay between these operations gets lower. IIRC, when testing against staging from my local machine, delay between portDown() and portUp() exceeded 1 second.
Also, to further speed up things, we may have to consider something faster, like directly managing the ports on the switches thru LabService, to avoid NB/Storm/Kafka/Floodlight traversal.

Reply:
Tried the async nb calls, but unfortunately it resulted in even slower port switches. Will investigate the FL switch possibility later